### PR TITLE
[TEST] getAllApplications API Endpoint

### LINKED ISSUE #407: [TEST] getAllApplications API Endpoint
### Is your feature request related to a problem? Please describe.
Implement a test suite that will cover all required functionality expected from the getAllApplications endpoint `/v1/applications`. For each test that doesn't pass, the controller method (along with its validation schema and swagger docs if the modifications require it) will be modified to include the required functionalities so that the test passes.

### Describe the solution you'd like
1. test that a request returns with the correct update response and status 200
2. test that a request with an unauthorized token returns with correct error response and status 401

### Describe alternatives you've considered
None

### Additional context
None

### PR DESCRIPTION
**Pull request by simply-alliv**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**

Fixes #407 

This PR implements a test suite that will cover all required functionality expected from the getAllApplications endpoint /v1/applications. For each test that doesn't pass, the controller method (along with its validation schema and swagger docs if the modifications require it) will be modified to include the required functionalities so that the test passes.

**description of Task to be completed?**

- [x] test that a request returns with the correct update response and status 200
- [x] test that a request with an unauthorized token returns with correct error response and status 401

**How should this be manually tested?**

- In your terminal, run `npm test`

**Any background context you want to provide?**

None

**What is the link to the issue on Github?**

[Issue #407](https://github.com/microapidev/comment-microapi/issues/407)

**Questions:**
None

### CODE REVIEW

File: controllers/applicationsController/getAllApplications.js [hunk 1]
```diff
@@ -17,20 +16,14 @@ const getAllApplications = async (req, res, next) => {
   try {
     const { organizationId } = req.token;
     if (!mongoose.Types.ObjectId.isValid(organizationId)) {
-      return next(new CustomError(400, "Invalid OrganizationID"));
-    }
-
-    const organization = await Organizations.find({ _id: organizationId });
-    if (!organization) {
-      return next(new CustomError(400, "Invalid organization"));
+      return next(new CustomError(422, "Invalid OrganizationID"));
```
Comment (dave-ok) [inline controllers/applicationsController/getAllApplications.js:19]: Looking at this again maybe we should change this to invalid organization token instead?

File: __test__/api/applications/getAllApplications.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    expect(res.status).toEqual(200);
+    expect(res.body.data).toEqual([
+      {
+        applicationId: String(global.application._id),
+        name: global.application.name,
+      },
+      {
+        applicationId: String(application._id),
+        name: application.name,
+      },
```
Comment (dave-ok) [inline __test__/api/applications/getAllApplications.test.js:44]: Neat trick!

File: controllers/applicationsController/getAllApplications.js [hunk 3]
```diff
@@ -17,20 +16,14 @@ const getAllApplications = async (req, res, next) => {
...
     }
```
Comment (dave-ok) [inline controllers/applicationsController/getAllApplications.js:20]: Why remove this?

File: controllers/applicationsController/getAllApplications.js (hunk 1 above)
Comment (simply-alliv) [reply controllers/applicationsController/getAllApplications.js:19]: We don't even need to know if the organization exists or not in this controller for it to perform its function. That job should be done by another.

File: controllers/applicationsController/getAllApplications.js (hunk 3 above)
Comment (simply-alliv) [reply controllers/applicationsController/getAllApplications.js:20]: As explained in the first review conversation.

File: __test__/api/applications/getAllApplications.test.js (hunk 2 above)
Comment (simply-alliv) [reply __test__/api/applications/getAllApplications.test.js:44]: Haha, thanks.

File: controllers/applicationsController/getAllApplications.js (hunk 1 above)
Comment (dave-ok) [reply controllers/applicationsController/getAllApplications.js:19]: What other function or controller checks this? An organization may have been removed and the organization token used before it expires, we may end up allowing such invalid requests through without this check

Comment (simply-alliv) [reply controllers/applicationsController/getAllApplications.js:19]: Can't the authentication middleware that runs check this before the controller method is called? That way, the check is done once for all applications endpoints and the code respects the DRY principle. If we add the check here then we will need to add the check in every application and admin endpoint.

Comment (dave-ok) [reply controllers/applicationsController/getAllApplications.js:19]: You are right. I guess we may need to use mongo hooks and middleware to ensure foreign relationships/references are maintained. So if e.g an org is deleted it's respective apps will be deleted as well

**Pull request merged**